### ISSUE TITLE
Simulate NOT IN operator

### ISSUE THREAD

**Issue by Mc128k** [ujjwalguptaofficial/JsStore#38]
Hi
I'm having difficulties simulating the NOT IN (...) operation.

As there is no negation operator for `In`, I thought that an AND operation with the same field would be fine, but it produces an error instead. This is the code I'm playing with:
```
    {
            From: 'Job',
            Where: [{
              dateExpiryStart: { '>': 0 },
              state: { '!=': 'Concluded' }
            }, {
              state: { '!=': 'WaitPickup' }
            }
            ],
    };
```

Any suggestion?

**Comment by ujjwalguptaofficial** [comment]
What's the error message ? Also Can you write your query in sql, so that I can simulate what will be equivalent query in jsstore.

**Comment by Mc128k** [comment]
No error, there just isn't a syntax that makes it possible in the documentation.

`SELECT * FROM Job WHERE dateExpiryStart > 0 AND state NOT IN ('Concluded', 'WaitPickup)`

Also I'm having problem identifying records with empty fields, but I'm still experimenting here

**Comment by ujjwalguptaofficial** [comment]
yeah i have not tested this query. I will add this.

Thanks

**Comment by ujjwalguptaofficial** [comment]
I was just guessing if this feature is there, equivalent qry will be something like that.

**Comment by Mc128k** [comment]
You mean the `Where: []`, right?
In order to use the last dev build (so I can also perform some testing), what should I do? I also see that you're working on a v2.0, should I use that?

**Comment by ujjwalguptaofficial** [comment]
Yeah you can use the version2 But its not completed yet - need to create docs, some testing etc. But yeah if you will start using, it will be great for me. You can let me know if there is any issue, so that I can fix it before publishing v2.

V2 has some little changes in syntax, and the way it is worked. So until i update the docs, it may be little tough to use. I have updated the examples folder, so you can figure it out - How to use.

Thanks

**Comment by ujjwalguptaofficial** [comment]
@Mc128k - can you try this query : 

 {
     from: 'Job',
     where: [{
         dateExpiryStart: {
             '>': 0
         },
         state: {
             '!=': 'Concluded'
         }
     }, {
         state: {
             '!=': 'WaitPickup'
         }
     }],
 };

I know you have tried it before, but there was some fixes in 2.0.5 . So it should work now. Let me know.

Thanks

**Comment by ujjwalguptaofficial** [comment]
I think - issue is still there. I was trying to produce and found the issue. I will fix it in next release.

Thanks

**Comment by Mc128k** [comment]
Yes, I found another selection issue after this, but I'll wait for the next release before posting

**Comment by ujjwalguptaofficial** [comment]
This issue has been resolved . Here is the idbstudio link - https://ujjwalguptaofficial.github.io/idbstudio/?query=select(%7B%0A%20%20%20%20from:%27Employees%27,%0A%20%20%20%20where:%20%5B%7B%0A%20%20%20%20%20%20notes:%20%7B%27like%27:%20%27%25from%25%27%7D,%0A%20%20%20%20%20%20state:%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%27!=%27:%20%27Concluded%27%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D,%20%7B%0A%20%20%20%20%20%20%20%20state:%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%27!=%27:%20%27WaitPickup%27%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%5D%0A%7D)&db=Demo

@Mc128k - let me know .

Thanks

**Comment by ujjwalguptaofficial** [comment]
@Mc128k - Please let me know another selection issue that you found. If the issue is not related to current issue then fire another issue.

**Comment by ujjwalguptaofficial** [comment]
@Mc128k - are we good on this. Can you close the issue if everything good ?

Thanks

**Comment by ujjwalguptaofficial** [comment]
Closing.

**Comment by rushi253095** [comment]
>  '!=': 'WaitPickup'

It will works in my problem thanks.

**Comment by rushi253095** [comment]
> @Mc128k - can you try this query :
> 
> ```
> {
>     from: 'Job',
>     where: [{
>         dateExpiryStart: {
>             '>': 0
>         },
>         state: {
>             '!=': 'Concluded'
>         }
>     }, {
>         state: {
>             '!=': 'WaitPickup'
>         }
>     }],
> };
> ```
> 
> I know you have tried it before, but there was some fixes in 2.0.5 . So it should work now. Let me know.
> 
> Thanks

It will works fine
